test/mocks/dlfcn_mock.cpp

```diff
@@ -0,0 +1,65 @@
+#include "mocks/dlfcn_mock.h"
```

Comment:
Can you add a file comment summarizing why this is needed? Especially include why the `extern "C"` stuff is needed, copy-paste of your PR description with a few changes should be fine.